servo/connectors/kubernetes.py

```diff
@@ -2100,19 +2156,21 @@ async def handle_error(self, error: Exception, mode: "FailureMode") -> bool:
 
             elif mode == FailureMode.rollback:
                 await self.rollback(error)
-                return True
 
             elif mode == FailureMode.destroy:
                 await self.destroy(error)
-                return True
+
+            else:
+                # Trap any new modes that need to be handled
+                raise NotImplementedError(
+                    f"missing error handler for failure mode '{mode}'"
+                ) from error
+
+            raise error # Always communicate errors to backend unless ignored
```

Comment:
Isn't this code unreachable?

Reply:
Its hit when FailureMode is rollback or destroy